lib/Driver/DarwinToolChains.cpp

```diff
@@ -321,10 +321,28 @@ toolchains::Darwin::addArgsToLinkARCLite(ArgStringList &Arguments,
 
 void toolchains::Darwin::addLTOLibArgs(ArgStringList &Arguments,
                                        const JobContext &context) const {
-  llvm::SmallString<128> LTOLibPath;
-  if (findXcodeClangLibPath("libLTO.dylib", LTOLibPath)) {
+  if (!context.OI.LibLTOPath.empty()) {
+    // Check for user-specified LTO library.
     Arguments.push_back("-lto_library");
-    Arguments.push_back(context.Args.MakeArgString(LTOLibPath));
+    Arguments.push_back(context.Args.MakeArgString(context.OI.LibLTOPath));
+  } else {
+    // Check for relative libLTO.dylib. This would be the expected behavior in an
+    // Xcode toolchain.
+    StringRef P = llvm::sys::path::parent_path(getDriver().getSwiftProgramPath());
```

Comment:
I think `P` is going to be `<toolchain root>/usr/bin/` here, so we'll end up checking for libLTO in `<toolchain root>/usr/bin/lib` instead of `<toolchain root>/usr/lib`. I can't think of a good way to write a lit test for this unfortunately.

Reply:
Thanks for pointing this out. Fixed in latest commit.